spdx/writers/rdf.py

```diff
@@ -982,8 +982,9 @@ def create_doc(self):
             self.graph.add((doc_node, self.spdx_namespace.name, doc_name))
         return doc_node
 
-    def write(self):
-        doc_node = self.create_doc()
+    def write(self, doc_node=None):
+        if not doc_node:
+            doc_node = self.create_doc()
```

Comment:
Can you explain in which cases it would be useful to pass the `doc_node` into this function?

Moreover, I don't like adding functionality that is never used. Depending on the use case, we'd have to make sure there are some paths that actually pass a `doc_node` to this function

Reply:
Thinking about this again, maybe the issue is that I don't have a good understanding how people use the Python tools. I wrote the previous comment with the mindset that a CLI is provided and should always be the entrypoint. But if the tools are used as a library and e.g. the `write` method is called directly, that does not apply (I need to get used to the absence of access modifiers in Python again - I automatically interpreted the `write` method as private :sweat_smile: ).

I am still curious about the use case for passing in a doc node here, but my comment about missing uses can be ignored.

Reply:
@nicoweidner : Probably both use cases are valid. I personally want to use it as library to parse, convert and write SPDX files within my own python construct. The CLI use case is probably used more often, as more inexperienced users use it, but please do not forget the library use case :-)

Reply:
@jblu42 Since you commented here: Do you see the use case for adding this option? The original author seems to be unresponsive (I don't mean to blame here, it's been almost a year...), and there is no description. I'd like to have at least some understanding of the use case before merging this

Reply:
@nicoweidner : I have not used the writers (yet), I can quickly describe my use case: I use the python tools to parse and extract information from an SPDX SBOM within a larger python script where the input can be TV or other formats and to gather other SBOM data from different sources and files. In the future I might also write out information in spdx format again using this library.

I have not really deeply checked the code, my requirement would be that I can set the information in the SDPX header that is here contained in the doc_node, if this is required to set it, then this commit makes sense, if I can define it via attributes of the document in general, this does not make sense to set it at the writer stage.

Reply:
Hi all, first of all sorry for my late replay.
When this PR is created this tools is used as a library and the document is created outside using the same `create_doc()` function. The issue there is because we can't change anything in the `doc_node` object fields.

Reply:
Hi @quaresmajose, thanks for coming back and chiming in! I realize this project was rather inactive for a while, so it's nice to see that people are still interested and coming back to old PRs.

Thanks for the explanation of the usage (also @jblu42). I'm fine with adding this functionality in, but the branch needs to be rebased on `main`.
And, as Philippe pointed out before, adding a small unit test using the functionality may help avoid someone accidentally removing it (using myself as an example here :sweat_smile: )

Reply:
Understand that the unit test is very useful, when I submit the PR didn't existed any test in the `writers/rdf.py` so I can't take any example in code. I haven't used this project for a while so I'm a little out of context and don't feel capable nor do I understand enough to write a test, if anyone wants to use or modify this change fell free.

Reply:
@quaresmajose No problem, I'll try to add a test soon!